sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -516,7 +516,7 @@ case class MapEntries(child: Expression) extends UnaryExpression with ExpectsInp
   examples = """
     Examples:
       > SELECT _FUNC_(map(1, 'a', 2, 'b'), map(2, 'c', 3, 'd'));
-       {1:"a",2:"c",3:"d"}
+       {1:"a",2:"b",2:"c",3:"d"}
```

Comment:
which spark version did you use? Duplicated entries were fixed and it shows:

```sql
spark-sql> SELECT map_concat(map(1, 'a', 2, 'b'), map(2, 'c', 3, 'd'));
{1:"a",2:"c",3:"d"}
```

and 

```scala
scala> spark.sql("SELECT map_concat(map(1, 'a', 2, 'b'), map(2, 'c', 3, 'd'))").show(20, false)
+--------------------------------------------+
|map_concat(map(1, a, 2, b), map(2, c, 3, d))|
+--------------------------------------------+
|[1 -> a, 2 -> c, 3 -> d]                    |
+--------------------------------------------+
```

in the current master.

Reply:
You are right. Looks like I've tested on v2.4.0 because my SPARK_HOME targeted to the old repo, and the logic was changed towards to exclude duplicates in https://github.com/apache/spark/pull/23124 .
Now I'm checking on the latest master and have the same results as yours.
Thank you!